web/src/navigator.rs

```diff
@@ -388,9 +368,22 @@ impl NavigatorBackend for WebNavigatorBackend {
             let window = web_sys::window().expect("window()");
             let fetchval = JsFuture::from(window.fetch_with_request(&web_request))
                 .await
-                .map_err(|_| ErrorResponse {
-                    url: url.to_string(),
-                    error: Error::FetchError("Got JS error".to_string()),
+                .map_err(|_| {
+                    if url.scheme() == "file" {
+                        match create_specific_fetch_error(
+                            "WASM target can't fetch local URL",
+                            url.as_str(),
+                            "",
+                        ) {
+                            Err(e) => e,
+                            Ok(_) => unreachable!("create_specific_fetch_error never returns Ok"),
```

Comment:
`create_specific_fetch_error` should return `E` instead of `Result<S, E>` then, and callers should wrap it with `Err`